assembly/src/ast/tests.rs

```diff
@@ -1288,3 +1289,45 @@ end
 
     assert_eq!(&formatted, expected);
 }
+
+#[test]
+fn test_words_roundtrip_formatting() {
+    let source = "\
+const.A=0x0200000000000000020000000000000002000000000000000200000000000000
+const.B=[2,2,2,2]
+begin
+    push.0x0200000000000000020000000000000002000000000000000200000000000000.2
+    push.A.2
+    push.B.2
+    push.2.2.2.2
+end
+";
+
+    let context = TestContext::default();
+    let source = source_file!(&context, source);
+
+    let module = Module::parse(
+        LibraryPath::new_from_components(LibraryNamespace::Exec, []),
+        ModuleKind::Executable,
+        source,
+    )
+    .unwrap_or_else(|err| panic!("{err}"));
+
+    let formatted = module.to_string();
+    let expected = "\
+begin
+    push.[2,2,2,2]
```

Comment:
nit: Would it not be more consistent to render this as a series of `push.2`? Basically semantically `push.2.2.2.2` is equivalent to `push.A` or `push.B`, but the first is "destructured" while the second isn't.